pkg/keys/ecdsa.go

```diff
@@ -4,64 +4,56 @@ import (
 	"bytes"
 	"crypto/ecdsa"
 	"crypto/elliptic"
+	"crypto/rand"
 	"crypto/sha256"
-	"encoding/asn1"
+	"crypto/x509"
 	"encoding/json"
+	"encoding/pem"
 	"errors"
 	"fmt"
 	"io"
-	"math/big"
 
 	"github.com/theupdateframework/go-tuf/data"
 )
 
 func init() {
 	VerifierMap.Store(data.KeyTypeECDSA_SHA2_P256, NewEcdsaVerifier)
+	SignerMap.Store(data.KeyTypeECDSA_SHA2_P256, newEcdsaSigner)
 }
 
 func NewEcdsaVerifier() Verifier {
-	return &p256Verifier{}
+	return &EcdsaVerifier{}
 }
 
-type ecdsaSignature struct {
-	R, S *big.Int
+func newEcdsaSigner() Signer {
+	return &ecdsaSigner{}
 }
 
-type p256Verifier struct {
-	PublicKey data.HexBytes `json:"public"`
+type EcdsaVerifier struct {
+	PublicKey *PKIXPublicKey `json:"public"`
+	ecdsaKey  *ecdsa.PublicKey
 	key       *data.PublicKey
 }
 
-func (p *p256Verifier) Public() string {
-	return p.PublicKey.String()
+func (p *EcdsaVerifier) Public() string {
+	r, _ := x509.MarshalPKIXPublicKey(p.ecdsaKey)
```

Comment:
Why are we not checking for an error here?

Reply:
Hmm this was done because this assumes by the time a verifier is created (either done through unmarshalling or explcit construction) that it is well-formed. It looks like in `UnmarhsalPublicKey` that is not verified. Adding that, and adding a comment on the error handling here.

Reply:
If that's the case, I'd like to see an assertion that the error is nil.

Reply:
Since Go doesn't have assertions, I can go nuclear and use `panic(err)`: I think we should do that until the issue I filed adding graceful error handling here is resolved.

Reply:
Where is the 🔥 reaction emoji when you need it...